modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/Amount.scala

```diff
@@ -1,31 +1,59 @@
 package org.tessellation.rosetta.domain
 
+import cats.implicits.catsSyntaxEitherId
+
 import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
-import org.tessellation.rosetta.domain.currency.Currency
+import org.tessellation.rosetta.domain.currency.{Currency, DAG}
+import org.tessellation.schema.transaction.{TransactionAmount, TransactionFee}
 
-import derevo.cats.eqv
+import derevo.cats.{eqv, show}
 import derevo.circe.magnolia._
 import derevo.derive
 import eu.timepit.refined.api.Refined
 import eu.timepit.refined.auto._
 import eu.timepit.refined.boolean.Not
 import eu.timepit.refined.cats._
 import eu.timepit.refined.generic.Equal
+import eu.timepit.refined.numeric.{NonNegative, Positive}
+import eu.timepit.refined.refineV
 import io.circe.refined._
 import io.estatico.newtype.macros.newtype
 
 object amount {
-  @derive(eqv, customizableDecoder, customizableEncoder)
+  type AmountValuePredicate = Not[Equal[0L]]
+  type AmountValueRefined = Long Refined AmountValuePredicate
+
+  @derive(eqv, decoder, encoder, show)
+  @newtype
+  case class AmountValue(value: AmountValueRefined) {
+    def isNegative: Boolean = value < 0L
+    def isPositive: Boolean = value > 0L
+
+    def negate: Either[String, AmountValue] =
+      if (value.value == Long.MinValue)
+        "Long.MinValue cannot be negated".asLeft
+      else
+        refineV[AmountValuePredicate](-value.value).map(rv => AmountValue(rv))
```

Comment:
If at this point we know that the negation must work, because we checked the value is not `Long.MinValue` (we need to be sure it's the only special case) I think we can use the unsafe method here and return as `Right`. In the end we can have something like `negate: Either[MinLongNegationError, AmountValue]`